pcf/particle/aws/glacier/glacier_vault.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from botocore.exceptions import ClientError
+from pcf.core.aws_resource import AWSResource
+from pcf.core import State
+from pcf.util import pcf_util
+
+logger = logging.getLogger(__name__)
+
+class GlacierVault(AWSResource):
+    """
+    This is the implementation of Amazon's Glacier service. Additional functions that are callable on this particle
+    are upload_archive, initiate_job, list_jobs, delete_archive, delete_job, add_tags_to_vault, list_tags_for_vault
+    """
+    flavor = "glacier"
+
+    state_lookup = {
+        "missing": State.terminated,
+        "active": State.running,
+        "inactive": State.terminated
+    }
+
+    equivalent_states = {
+        State.running: 1,
+        State.stopped: 0,
+        State.terminated: 0
+    }
+
+    def __init__(self, particle_definition):
+        super(GlacierVault, self).__init__(
+            particle_definition=particle_definition,
+            resource_name="glacier",
+        )
+        self.vault_name = self.desired_state_definition.get("vaultName")
+        self.account_id = self.desired_state_definition.get("accountId", "-")
+
+    def _terminate(self):
+        """
+        Deletes the Glacier vault if vault is empty
+
+        Returns:
+             response of boto3 delete_vault
+        """
+        return self.client.delete_vault(vaultName=self.vault_name)
+
+    def _start(self):
+        """
+        Creates the Glacier vault
+
+        Returns:
+             response of boto3 create_vault
+        """
+        response = self.client.create_vault(
+            vaultName=self.vault_name,
+            accountId=self.account_id
+        )
+
+        if self.custom_config.get("Tags"):
+            tags = self.custom_config.get("Tags")
+
+            self.client.add_tags_to_vault(
+                vaultName=self.vault_name,
+                Tags=tags
+            )
+        return response
+
+
+    def _stop(self):
+        """
+        Glacier does not have a stopped state so it calls terminate.
+        """
+        return self.terminate()
+
+    def get_status(self):
+        """
+        Determines if the vault exists
+
+        Returns:
+             status (dict)
+        """
+
+        try:
+            vault_object = self.client.describe_vault(
```

Comment:
you should return this. this is the status.

Reply:
The response of `describe_vault()` does not include a `status` key. I followed the similar approach as the [S3 Particle](https://github.com/capitalone/Particle-Cloud-Framework/blob/1b3fb23edf6454c5db6b78b49751b266d4829529/pcf/particle/aws/s3/s3_bucket.py#L72) to handle this case.

Reply:
s3 is a unique case since it does not have a `describe_bucket()` or even a `get_bucket` so there is no way to run this call. all the other particles follow this convention of setting the `current_state_definition` to the describe obj call.

Reply:
so you should return `none` if missing else `vault object`.

ex.

```
             vault_object = self.client.describe_vault( 
                vaultName=self.vault_name,
                accountId=self.account_id
            )
        except ClientError: 
            return {}
        return vault_object
```